patches/openwrt/0012-ipq40xx-add-support-for-the-ZyXEL-NBG6617.patch

```diff
@@ -0,0 +1,1023 @@
+new file mode 100644
+index 0000000000000000000000000000000000000000..7c513a6ad8ec817bf388c67e1181f08982888fa5
+--- /dev/null
++++ b/target/linux/ipq40xx/patches-4.14/303-spi-nor-enable-4B-opcodes-for-mx25l25635f.patch
+@@ -0,0 +1,64 @@
++Index: linux-4.14.50/drivers/mtd/spi-nor/spi-nor.c
++===================================================================
++--- linux-4.14.50.orig/drivers/mtd/spi-nor/spi-nor.c
+++++ linux-4.14.50/drivers/mtd/spi-nor/spi-nor.c
++@@ -1025,6 +1025,7 @@ static const struct flash_info spi_nor_i
++ 	{ "mx25l12805d", INFO(0xc22018, 0, 64 * 1024, 256, 0) },
++ 	{ "mx25l12855e", INFO(0xc22618, 0, 64 * 1024, 256, 0) },
++ 	{ "mx25l25635e", INFO(0xc22019, 0, 64 * 1024, 512, SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+++	{ "mx25l25635f", INFO(0xc22019, 0, 64 * 1024, 512, SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
++ 	{ "mx25u25635f", INFO(0xc22539, 0, 64 * 1024, 512, SECT_4K | SPI_NOR_4B_OPCODES) },
++ 	{ "mx25l25655e", INFO(0xc22619, 0, 64 * 1024, 512, 0) },
++ 	{ "mx66l51235l", INFO(0xc2201a, 0, 64 * 1024, 1024, SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
++@@ -1189,11 +1190,12 @@ static const struct flash_info spi_nor_i
++ 	{ },
++ };
++ 
++-static const struct flash_info *spi_nor_read_id(struct spi_nor *nor)
+++static const struct flash_info *spi_nor_read_id(struct spi_nor *nor,
+++						const char *name)
++ {
++ 	int			tmp;
++ 	u8			id[SPI_NOR_MAX_ID_LEN];
++-	const struct flash_info	*info;
+++	const struct flash_info	*info, *first_match = NULL;
++ 
++ 	tmp = nor->read_reg(nor, SPINOR_OP_RDID, id, SPI_NOR_MAX_ID_LEN);
++ 	if (tmp < 0) {
++@@ -1204,10 +1206,16 @@ static const struct flash_info *spi_nor_
++ 	for (tmp = 0; tmp < ARRAY_SIZE(spi_nor_ids) - 1; tmp++) {
++ 		info = &spi_nor_ids[tmp];
++ 		if (info->id_len) {
++-			if (!memcmp(info->id, id, info->id_len))
++-				return &spi_nor_ids[tmp];
+++			if (!memcmp(info->id, id, info->id_len)) {
+++				if (!name || !strcmp(name, info->name))
+++					return info;
+++				if (!first_match)
+++					first_match = info;
+++			}
++ 		}
++ 	}
+++	if (first_match)
+++		return first_match;
++ 	dev_err(nor->dev, "unrecognized JEDEC id bytes: %02x, %02x, %02x\n",
++ 		id[0], id[1], id[2]);
++ 	return ERR_PTR(-ENODEV);
++@@ -2667,7 +2675,7 @@ int spi_nor_scan(struct spi_nor *nor, co
++ 		info = spi_nor_match_id(name);
++ 	/* Try to auto-detect if chip name wasn't specified or not found */
++ 	if (!info)
++-		info = spi_nor_read_id(nor);
+++		info = spi_nor_read_id(nor, NULL);
++ 	if (IS_ERR_OR_NULL(info))
++ 		return -ENOENT;
++ 
++@@ -2678,7 +2686,7 @@ int spi_nor_scan(struct spi_nor *nor, co
++ 	if (name && info->id_len) {
++ 		const struct flash_info *jinfo;
++ 
++-		jinfo = spi_nor_read_id(nor);
+++		jinfo = spi_nor_read_id(nor, name);
++ 		if (IS_ERR(jinfo)) {
++ 			return PTR_ERR(jinfo);
++ 		} else if (jinfo != info) {
```

Comment:
This must be merged in OpenWrt.

Reply:
Merged in master as well but (sadly) not in release https://github.com/openwrt/openwrt/commit/82618062cf7ed6b40d2c52c6f6b96364888ffda6

Reply:
I'll merge this to 18.06 after the final release (which is scheduled for next weekend), let's remove it from this PR for now.

Reply:
Ok, i'm fine with that :+1: